### PR TITLE
common: add uninstall target

### PR DESCRIPTION
**Pull request by wlemkows**
This change is [Reviewable](https://reviewable.io/reviews/pmem/miniasync/27)

### CODE REVIEW

**Review by lukaszstolarczuk — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/miniasync/27)* status: 0 of 4 files reviewed, 2 unresolved discussions (waiting on @kswiecicki and @wlemkows)

___
*[CMakeLists.txt, line 125 at r1](https://reviewable.io/reviews/pmem/miniasync/27#-MqZPP5g425rBqo3xGHl:-MqZPP5g425rBqo3xGHm:b49yvi6) ([raw file](https://github.com/pmem/miniasync/blob/7604f7c356bfc09978485215da6edbb76cc8ab06/CMakeLists.txt#L125)):*
> ```CMake
> 
> configure_file(
> 	"${CMAKE_SOURCE_DIR}/cmake/cmake_uninstall.cmake.in"
> ```

MINIASYNC_ROOT_DIR :wink: 
___
*[.github/workflows/on_pull_request.yml, line 30 at r1](https://reviewable.io/reviews/pmem/miniasync/27#-MqZPYGO8OigwXINttPN:-MqZPYGO8OigwXINttPO:bdlyzof) ([raw file](https://github.com/pmem/miniasync/blob/aff8486c6bd71335c7324e59a9cc87668c233788/.github/workflows/on_pull_request.yml#L30)):*
> ```YAML
>                  "N=2 OS=ubuntu OS_VER=21.04 TYPE=normal CC=clang PUSH_IMAGE=1",
>                  "N=3 OS=fedora OS_VER=34    TYPE=normal CC=gcc   PUSH_IMAGE=1"]
>                  #"N=4 OS=fedora OS_VER=34    TYPE=normal CC=clang AUTO_DOC_UPDATE=1"]
> ```

in this PR...?

**Review by wlemkows — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/miniasync/27)* status: 0 of 4 files reviewed, 1 unresolved discussion (waiting on @kswiecicki and @lukaszstolarczuk)

___
*[CMakeLists.txt, line 125 at r1](https://reviewable.io/reviews/pmem/miniasync/27#-MqZPP5g425rBqo3xGHl:-MqmqkQo1f-BPWSW1ww-:b-896fix) ([raw file](https://github.com/pmem/miniasync/blob/7604f7c356bfc09978485215da6edbb76cc8ab06/CMakeLists.txt#L125)):*
<details><summary><i>Previously, lukaszstolarczuk (Łukasz Stolarczuk) wrote…</i></summary><blockquote>

MINIASYNC_ROOT_DIR :wink: 
</blockquote></details>

Done.
___
*[.github/workflows/on_pull_request.yml, line 30 at r1](https://reviewable.io/reviews/pmem/miniasync/27#-MqZPYGO8OigwXINttPN:-MqmvJ3VEvCJ4-Xnx38J:bfy6g3o) ([raw file](https://github.com/pmem/miniasync/blob/aff8486c6bd71335c7324e59a9cc87668c233788/.github/workflows/on_pull_request.yml#L30)):*
<details><summary><i>Previously, lukaszstolarczuk (Łukasz Stolarczuk) wrote…</i></summary><blockquote>

in this PR...?
</blockquote></details>

Done. (separate commit)

**Review by lukaszstolarczuk — COMMENTED** [review]
:+1:

> Reviewed 2 of 4 files at r1, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/pmem/miniasync/27#-:-Mqmyjs46BcKwhg6GJpd:bb6mt9b)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @kswiecicki)

File: cmake/cmake_uninstall.cmake.in [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# From: https://cmake.org/Wiki/CMake_FAQ
```
Comment (pbalcer) [inline cmake/cmake_uninstall.cmake.in:1]: it seems the code changed a bit (https://gitlab.kitware.com/cmake/community/-/wikis/FAQ#can-i-do-make-uninstall-with-cmake). Can we use the updated version?

**Review by wlemkows — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/miniasync/27)* status: 2 of 4 files reviewed, 1 unresolved discussion (waiting on @kswiecicki, @lukaszstolarczuk, and @pbalcer)

___
*[cmake/cmake_uninstall.cmake.in, line 1 at r3](https://reviewable.io/reviews/pmem/miniasync/27#-MqZK8etLCh_IZV8NWyi-r3-1:-MqsKYkuEOIQrh8YFbZE:b-896fix) ([raw file](https://github.com/pmem/miniasync/blob/d6264ba9ca071c12ab422cd00eb5e4c2da759fd5/cmake/cmake_uninstall.cmake.in#L1)):*
<details><summary><i>Previously, pbalcer (Piotr Balcer) wrote…</i></summary><blockquote>

it seems the code changed a bit (https://gitlab.kitware.com/cmake/community/-/wikis/FAQ#can-i-do-make-uninstall-with-cmake). Can we use the updated version?
</blockquote></details>

Done.

**Review by pbalcer — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/pmem/miniasync/27#-:-MqsWIAn4IzDXUZCgS1p:bnfp4nl)* status: 2 of 4 files reviewed, all discussions resolved (waiting on @kswiecicki and @lukaszstolarczuk)

**Pull request merged**